x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/X-StdDeviationAggregator.java.st

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.ObjectArray;
+import org.elasticsearch.compute.ann.Aggregator;
+import org.elasticsearch.compute.ann.GroupingAggregator;
+import org.elasticsearch.compute.ann.IntermediateState;
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.DoubleBlock;
+import org.elasticsearch.compute.data.IntVector;
+import org.elasticsearch.compute.operator.DriverContext;
+import org.elasticsearch.core.Releasables;
+
+/**
+ * A standard deviation aggregation definition for $type$.
+ * This class is generated. Edit `X-StdDeviationAggregator.java.st` instead.
+ */
+@Aggregator(
+    {
+        @IntermediateState(name = "mean", type = "DOUBLE"),
+        @IntermediateState(name = "m2", type = "DOUBLE"),
+        @IntermediateState(name = "count", type = "LONG") }
+)
+@GroupingAggregator
+public class StdDeviation$Type$Aggregator {
+
+    public static StdDeviation$Type$State initSingle() {
+        return new StdDeviation$Type$State();
+    }
+
+    public static void combine(StdDeviation$Type$State state, $type$ value) {
+        state.add(value);
+    }
+
+    public static void combineIntermediate(StdDeviation$Type$State state, double mean, double m2, long count) {
+        state.combine(mean, m2, count);
+    }
+
+
+    public static void evaluateIntermediate(StdDeviation$Type$State state, DriverContext driverContext, Block[] blocks, int offset) {
+        assert blocks.length >= offset + 3;
+        BlockFactory blockFactory = driverContext.blockFactory();
+        blocks[offset + 0] = blockFactory.newConstantDoubleBlockWith(state.mean(), 1);
+        blocks[offset + 1] = blockFactory.newConstantDoubleBlockWith(state.m2(), 1);
+        blocks[offset + 2] = blockFactory.newConstantLongBlockWith(state.count(), 1);
+    }
+
+    public static Block evaluateFinal(StdDeviation$Type$State state, DriverContext driverContext) {
+        final long count = state.count();
+        final double m2 = state.m2();
+        if (count == 0 || Double.isFinite(m2) == false) {
+            return driverContext.blockFactory().newConstantNullBlock(1);
```

Comment:
If the result is infinity or NaN I set it to return null, but I'm not sure if there should be a warning or something similar printed (or where that would best be done)?